lib/Sema/TypeCheckDecl.cpp

```diff
@@ -2215,20 +2216,39 @@ InterfaceTypeRequest::evaluate(Evaluator &eval, ValueDecl *D) const {
       }
     }
 
-    if (!PD->getTypeRepr())
-      return Type();
+    if (PD->getTypeRepr()) {
+      return validateParameterType(PD);
+    }
+
+    // Try type checking the closure if it hasn't.
+    if (auto *closure = dyn_cast<ClosureExpr>(PD->getDeclContext())) {
```

Comment:
Whoa, this is a whole lot of work to do in the interface type request. If you need this information, you may as well just typecheck the closure up front.

Reply:
I can remove this block because it turned out that this branch is actually unreachable. `typeCheckASTNodeAtLoc()` type checks surrounding closures anyway.
https://github.com/apple/swift/blob/c4095fc074aba6e9eb6dca26c2d0be770dc4fa71/lib/Sema/TypeCheckStmt.cpp#L1960-L1968

But I don't really understand your argument here. Could you clarify your concern with this on-demand type checking for getting interface types of the decl?

Reply:
Removed this block from the P.